azurerm/helpers/azure/batch_pool.go

```diff
@@ -518,3 +518,34 @@ func ValidateAzureRMBatchPoolName(v interface{}, k string) (warnings []string, e
 
 	return warnings, errors
 }
+
+// ExpandMetaData expands Batch pool metadata
+func ExpandMetaData(input map[string]interface{}) *[]batch.MetadataItem {
+	output := []batch.MetadataItem{}
+
+	for k, v := range input {
+		name := k
+		value := v.(string)
+		output = append(output, batch.MetadataItem{
+			Name:  &name,
+			Value: &value,
+		})
+	}
+
+	return &output
+}
+
+// FlattenMetaData flattens a Batch pool metadata
+func FlattenMetaData(metadatas *[]batch.MetadataItem) map[string]interface{} {
```

Comment:
since this package contains helpers for all different libraries - could we include the service name in the method:

```suggestion
func FlattenBatchMetaData(metadatas *[]batch.MetadataItem) map[string]interface{} {
```

Reply:
done